src/ResourceManager/DataFactoryV2/Commands.DataFactoryV2/IntegrationRuntimes/SetAzureDataFactoryIntegrationRuntimeCommand.cs

```diff
@@ -144,6 +152,15 @@ public override void ExecuteCmdlet()
             this.ByResourceId();
             this.ByIntegrationRuntimeObject();
 
+            if (AuthKey != null && !string.IsNullOrWhiteSpace(SharedIntegrationRuntimeResourceId))
```

Comment:
Same comment here.  Instead of doing runtime checks, you should do parameter set validation.  Here, you would create a new parameter set where AuthKey is not included and SharedIntegrationRuntimeResourceId is included.

Reply:
@zhangyd2015

Reply:
I am facing the same issue with `RemoveAzureDataFactoryIntegrationRuntimeCommand` above.

Reply:
There should be 6 parameter sets.  The first three are identical to what you had before this parameter added (and the default parameter set should be one of these parameter sets).  The other three should have -AuthKey removed, and should have -SharedIntegrationRuntimeResourceId as a **mandatory** parameter.

Reply:
Done.